tree/ntuple/v7/inc/ROOT/RField/RFieldSequenceContainer.hxx

```diff
@@ -287,6 +289,22 @@ public:
    }
 };
 
+/// The same memory and on-disk layout than an std::vector but without an assigned type name.
+/// Analogous to the RRecordField acting as an untyped class, this field is an untyped sequence collection.
+class RSequenceCollectionField final : public RVectorField {
```

Comment:
Wouldn't it make more sense to have it the other way around, ie `RVectorField` inherit from `RSequenceCollectionField` that takes a `type` argument which defaults to `""`?

Reply:
Perhaps, I have no strong opinion on that. See [commit](https://github.com/root-project/root/pull/16478/commits/9843e1569a4f9676bc169d8a590adb9db97cc56e).

Reply:
I don't have a strong opinion either. What I had in mind was to get rid of the `untyped` parameter like https://github.com/hahnjo/root/commit/aed26844544f0c56885079b3d4257dd3c00590df. I believe this disentangles the classes even more and leaves each of them with a clear responsibility: `RSequenceCollectionField` takes a `std::vector` in memory and orchestrates serialization to disk, while `RVectorField`'s only job is to provide a proper `typeName`. I leave the decision to you (in case you agree with my changes, feel free to just amend them into your commit)